src/diamond/handler/cloudwatch.py

```diff
@@ -193,12 +209,24 @@ def process(self, metric):
                     str(metric.timestamp)
                 )
                 try:
-                    self.connection.put_metric_data(
-                        str(rule['namespace']),
-                        str(rule['name']),
-                        str(metric.value),
-                        timestamp, str(rule['unit']),
-                        {'InstanceId': self.instance_id})
+                    if self.autoscaling:
+                        self.log.debug(
+                            'Cloudwatch: Publishing metric to AutoScaling'
+                        )
+                        self.cloudwatch.put_metric_data(
+                            str(rule['namespace']),
+                            str(rule['name']),
+                            str(metric.value),
+                            timestamp, str(rule['unit']),
+                            {'AutoScalingGroupName': autoscaling_group})
```

Comment:
It seems to me that these statements are essentially identical, with the exception of this line. Perhaps it makes sense to DRY this up a little by doing something like:

```python
if self.autoscaling:
  target = {'AutoScalingGroupName': autoscaling_group}
else:
  target = {'InstanceId': self.instance_id}

self.cloudwatch.put_metric_data(
                            str(rule['namespace']),
                            str(rule['name']),
                            str(metric.value),
                            timestamp, str(rule['unit']),
                            target)
```

And you can add your debug logging about where we're publishing to the simpler `if`.